lib/notesclub_web/controllers/auth_controller.ex

```diff
@@ -0,0 +1,55 @@
+defmodule NotesclubWeb.AuthController do
+  use NotesclubWeb, :controller
+  plug Ueberauth
+
+  alias Notesclub.Accounts
+
+  def callback(%{assigns: %{ueberauth_failure: _fails}} = conn, _params) do
+    conn
+    |> put_flash(:error, "Failed to authenticate.")
+    |> redirect(to: "/")
+  end
+
+  def callback(%{assigns: %{ueberauth_auth: auth}} = conn, _params) do
+    case Accounts.get_by_github_id(auth.uid) do
+      nil ->
+        user_params = %{
+          username: auth.info.nickname,
+          github_id: auth.uid,
+          name: auth.info.name,
+          avatar_url: auth.info.urls.avatar_url,
+          bio: auth.extra.raw_info.user["bio"],
+          email: auth.info.email,
+          location: auth.info.location,
+          followers_count: auth.extra.raw_info.user["followers"],
+          last_login_at: DateTime.utc_now()
+        }
+
+        case Accounts.create_user(user_params) do
+          {:ok, user} ->
+            conn
+            |> put_session(:user_id, user.id)
+            |> redirect(to: "/")
+
+          {:error, _changeset} ->
+            conn
+            |> put_session(:user_id, nil)
+            |> put_flash(:error, "There was an issue creating your account.")
+            |> redirect(to: "/")
+        end
+
+      user ->
+        Accounts.update_user(user, %{last_login_at: DateTime.utc_now()})
```

Comment:
I suppose GitHub provides all params every time we log in, does it?
Then, we could update them (not tested):

```suggestion
  def callback(%{assigns: %{ueberauth_auth: auth}} = conn, _params) do
    user_params = %{
      username: auth.info.nickname,
      github_id: auth.uid,
      name: auth.info.name,
      avatar_url: auth.info.urls.avatar_url,
      bio: auth.extra.raw_info.user["bio"],
      email: auth.info.email,
      location: auth.info.location,
      followers_count: auth.extra.raw_info.user["followers"],
      last_login_at: DateTime.utc_now()
    }
    
    case Accounts.get_by_github_id(auth.uid) do
      nil ->
        case Accounts.create_user(user_params) do
          {:ok, user} ->
            conn
            |> put_session(:user_id, user.id)
            |> redirect(to: "/")

          {:error, _changeset} ->
            conn
            |> put_session(:user_id, nil)
            |> put_flash(:error, "There was an issue creating your account.")
            |> redirect(to: "/")
        end

      user ->
        Accounts.update_user(user, user_params)
```

Reply:
Yes, we can update on every login. Good idea to keep all information updated. I will do it.

Reply:
5667d9453c8bc049c59037a97fa9eb357e6a7f5d